app/controllers/spotlight/concerns/application_controller.rb

```diff
@@ -25,6 +25,64 @@ module ApplicationController
           end
         end
       end
+
+      def enabled_in_spotlight_view_type_configuration?(config, *args)
+        case
+        when config.respond_to?(:upstream_if) && !config.upstream_if.nil? && !blacklight_configuration_context.evaluate_configuration_conditional(config.upstream_if, config, *args)
+          false
+        when current_exhibit.nil? || self.is_a?(Spotlight::PagesController)
+          true
+        else
+          current_exhibit.blacklight_configuration.document_index_view_types.include? config.key.to_s
+        end
+      end
+
+      # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/MethodLength
+      def field_enabled?(field, *args)
+        case
+        when !field.enabled
+          false
+        when (field.respond_to?(:upstream_if) && !field.upstream_if.nil? && !blacklight_configuration_context.evaluate_configuration_conditional(field.upstream_if, field, *args))
+          false
+        when field.is_a?(Blacklight::Configuration::SortField) || field.is_a?(Blacklight::Configuration::SearchField)
+          field.enabled
+        when field.is_a?(Blacklight::Configuration::FacetField) || (self.is_a?(Blacklight::Catalog) && %w(edit show).include?(action_name))
+          field.show
+        else
+          field.send(document_index_view_type)
+        end
+      end
+      # rubocop:enable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/MethodLength
+
+      private
+
+      ##
+      # Get the current "view type" (and ensure it is a valid type)
+      #
+      # @param [Hash] the query parameters to check
+      # @return [Symbol]
+      def document_index_view_type
+        view_param = params[:view]
+        view_param ||= session[:preferred_view]
+        if view_param and document_index_views.keys.include? view_param.to_sym
+          view_param.to_sym
+        else
+          default_document_index_view_type
+        end
+      end
+
+      def document_index_views
+        blacklight_config.view.select do |k, config|
+          blacklight_configuration_context.evaluate_if_unless_configuration config
+        end
+      end
+
+      ##
+      # Get the default index view type
+      def default_document_index_view_type
+        document_index_views.select { |k,config| config.respond_to? :default and config.default }.keys.first || document_index_views.keys.first
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Use `&&` instead of `and`.<br>Space missing after comma.